src/popup/modules/UserInterface.js

```diff
@@ -108,8 +110,18 @@ const refreshQrCode = throttle(() => {
         CommonMessages.hideError();
     }
 
-    QrCreator.setTextInternal(text);
-    QrCreator.generate();
+    try {
+        QrCreator.setTextInternal(text);
+        QrCreator.generate();
+    } catch (e) {
+        // Error thrown from qrcodegen & kjua wrapper when code too long
+        if (e instanceof QrError.DataOverflowError) {
+            CommonMessages.showError("errorQrCodeOverflow");
+            Logger.logError("Data exceeds maximum size:", text.length);
```

Comment:
```suggestion
            Logger.logError("Maximum size of QR code data exceeded with", text.length, "characters.");
```

The difference is that we do not know whether this is the minimal maximum size, i.e. the maximum size can be 1000 chars, but generation obviously still fails for 2000 chars then. So this log message is just more accurat. 😄